mediasession.bs

```
@@ -263,8 +269,53 @@ session state">idle</a></code>.
 The <dfn attribute for="MediaSession"><code>kind</code></dfn>
 attribute must return the <a>media session</a>'s <a>kind</a>.
 
+The <dfn method for="MediaSession"><code>activate()</code></dfn> method, when
+invoked, must run these steps:
+
+<ol>
+  <li>
+    Let <var>media session</var> be the <a>context object</a>.
+  </li>
+  <li>
+    Let <var>promise</var> be a new promise.
+  </li>
+  <li>
+    Return <var>promise</var>, and run the remaining steps <a>in parallel</a>.
+  </li>
+  <li>
+    <a>Activate</a> <var>media session</var>.
+  </li>
+  <li>
+    If <a>activate</a> failed, reject <var>promise</var> with a
+    <code>TypeError</code>.
```

This look like it's the implementation of `requestAudioFocus()` in Android:
https://github.com/android/platform_frameworks_base/blob/3f453164ec884d26a556477027b430cb22a9b7e3/media/java/android/media/MediaFocusControl.java#L714

I'm not sure which failure cases would be reachable for us, but without digging deeper I'm not very confident that all of the failure cases correspond to a more high-priority component holding audio focus.

I think we should just go with an single catch-all exception until we have the APIs and the use cases to do otherwise. Whether that single exception is `TypeError` or some kind of `DOMException` is the question we need to answer to get this merged.